vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+# Avalanche Warp Messaging
+
+Avalanche Warp Messaging (AWM) provides a primitive for cross-subnet communication on the Avalanche Network.
+
+The Avalanche P-Chain provides an index of every Subnet's validator set on the Avalanche Network, including the BLS public key of each validator (as of the [Banff Upgrade](https://github.com/ava-labs/avalanchego/releases/v1.9.0)). AWM utilizes the weighted validator sets stored on the P-Chain to build a cross-subnet communication protocol between any two Subnets on the Avalanche Network.
+
+Any Virtual Machine (VM) on Avalanche can integrate Avalanche Warp Messaging to send and receive messages across Avalanche Subnets.
+
+## Background
+
+This README assumes familiarity with:
+
+- Avalanche P-Chain / [PlatformVM](../)
+- [ProposerVM](../../proposervm/README.md)
+- Basic familiarity with [BLS Multi-Signatures](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html)
+
+## BLS Multi-Signatures with Public-Key Aggregation
+
+Avalanche Warp Messaging utilizes BLS multi-signatures with public key aggregation in order to verify messages signed by another Subnet. When a validator joins a Subnet, the P-Chain records the validator's BLS public key and NodeID, as well as a proof of possession of the validator's BLS private key to defend against [rogue public-key attacks](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html#mjx-eqn-eqaggsame).
+
+AWM utilizes the validator set's weights and public keys to verify that an aggregate signature has sufficient weight signing the message from the source Subnet.
+
+BLS provides a way to aggregate signatures off chain into a single signature that can be efficiently verified on chain.
+
+## AWM Serialization
```

Comment:
For this section we're making our own table diagrams but markdown supports this via https://www.markdownguide.org/extended-syntax/#tables

I'm not sure if you meant to do this if you prefer the ASCII look but this is something else to consider

Reply:
This is following the style we've used here: https://docs.avax.network/reference/avalanchego/p-chain/txn-format. Happy to change it to a more standard format throughout our docs, but don't think that should be a requirement for this PR.